Modules/_sqlite/connection.c

```diff
@@ -88,11 +88,8 @@ int pysqlite_connection_init(pysqlite_Connection* self, PyObject* args, PyObject
     Py_CLEAR(self->statements);
     Py_CLEAR(self->cursors);
 
-    Py_INCREF(Py_None);
-    Py_XSETREF(self->row_factory, Py_None);
-
-    Py_INCREF(&PyUnicode_Type);
-    Py_XSETREF(self->text_factory, (PyObject*)&PyUnicode_Type);
```

Comment:
I don't think that these changes are correct. Py_XSETREF() clears the old value before setting the new value.

Reply:
But, this is in the `__init__` method? Isn't `self` uninitialised at this point?

Reply:
`self` was not created at this function, so it makes sense to treat it as non-empty. Although not seemingly useful, it's okay to call `__init__` on an already-created object. `__new__` is what's promised to be called just once, IIRC. That also explains why you have those 3 `Py_CLEAR` in lines 87-89.

Reply:
Technically in Python, you can call an ``__init__()`` multiple times. It should be safe to call it more than once.